### DISCUSSION TITLE
BUG: Collection hooks can't determine when document gets unpublished

### DISCUSSION THREAD

**Discussion by johannesschaffer** [payloadcms/payload#2627] [Feature Requests & Ideas]
In earlier versions of payload, the _status was either: "published", "draft", "unpublished". At some point the "unpublished" status got removed. I can't check for unpublished by `doc._status == 'draft' && prevDoc._status == 'published'`, since this would also triggger for a  normal draft save, when not unpublishing it.

**Comment by qbeauperin** [comment]
You could check whether this doc has a published version by doing something like this in your hook:

```typescript
const afterChange: CollectionAfterChangeHook = async ({ doc, req: { payload }, previousDoc }) => {
    if (doc._status === 'draft' && previousDoc._status === 'published') {
        const publishedDoc = await payload.findByID({
            collection: collectionConfig.slug,
            id: doc.id,
            draft: false,
        });
        if (!publishedDoc || publishedDoc._status === 'draft') {
            console.log('Document was unpublished');
        } else {
            console.log('Just the first draft after publishing')
        }
    }

    return doc;
}
```

**Reply by johannesschaffer** [reply]
Thanks, will try that...
Question: Why do check if `publishedDoc._status === 'draft'` if you specified `draft: false` in `payload.findByID`? Is it still possible it returns a draft version?

**Reply by qbeauperin** [reply]
From the quick test I did before posting this snippet, it looked like if the doc isn't published, `findById` will return the latest draft, even with `draft: false`. Not sure if that's by design or not though. 🤷‍♂️🤔
